beancount/utils/test_utils.py

```diff
@@ -187,11 +188,11 @@ def new_function(self):
         allowed = ("buffering", "encoding", "newline", "dir", "prefix", "suffix")
         if any(key not in allowed for key in kwargs):
             raise ValueError("Invalid kwarg to docfile_extra")
-        with tempfile.NamedTemporaryFile("w", **kwargs) as file:
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_file_path = Path(temp_dir, "temp.txt")
             text = contents or function.__doc__
-            file.write(textwrap.dedent(text))
-            file.flush()
-            return function(self, file.name)
+            temp_file_path.write_text(textwrap.dedent(text), encoding="utf-8")
```

Comment:
This now silently ignores the `buffering`, `encoding`, `newline`, `dir`, `prefix`, and `suffix` arguments that can be passed to the `docfile_extra()` decorator. If these are not used anywhere in the code, support for them can most likely be removed, but ignoring them silently is not ok.

Reply:
is it ok now?